include/bluetooth/mesh/gen_dtt_srv.h

```diff
@@ -134,8 +131,11 @@ static inline bool
 bt_mesh_dtt_srv_transition_get(struct bt_mesh_model *model,
 			       struct bt_mesh_model_transition *transition)
 {
-	struct bt_mesh_dtt_srv *srv =
-		bt_mesh_dtt_srv_get(bt_mesh_model_elem(model));
+	struct bt_mesh_dtt_srv *srv = NULL;
+
+#ifdef CONFIG_BT_MESH_DTT_SRV
+	srv = bt_mesh_dtt_srv_get(bt_mesh_model_elem(model));
+#endif
```

Comment:
Just to avoid unnecessary initialization if compiled with server support and keep compiling the whole code even if server support is not enabled.
```suggestion
	struct bt_mesh_dtt_srv *srv;

	if (IS_ENABLED(CONFIG_BT_MESH_DTT_SRV)) {
		srv = bt_mesh_dtt_srv_get(bt_mesh_model_elem(model));
	} else {
		srv = NULL;
	}
```